File: android/zzt_qrcode_java/src/main/java/xyz/zhuzeitou/qrcode/LogLevel.java
package xyz.zhuzeitou.qrcode;

/** Log severity emitted by the native QR code library. */
public enum LogLevel {
    VERBOSE(0),
    DEBUG(1),
    INFO(2),
    WARN(3),
    ERROR(4);

    private final int nativeValue;

    LogLevel(int nativeValue) {
        this.nativeValue = nativeValue;
    }

    int nativeValue() {
        return nativeValue;
    }

    static LogLevel fromNativeValue(int value) {
        switch (value) {
            case 0:
                return VERBOSE;
            case 1:
                return DEBUG;
            case 2:
                return INFO;
            case 3:
                return WARN;
            case 4:
                return ERROR;
            default:
                return null;
        }
    }
}

File: android/zzt_qrcode_kotlin/src/main/java/xyz/zhuzeitou/qrcode/LogLevel.kt
package xyz.zhuzeitou.qrcode

enum class LogLevel(internal val nativeValue: Int) {
    VERBOSE(0),
    DEBUG(1),
    INFO(2),
    WARN(3),
    ERROR(4);

    companion object {
        internal fun fromNativeValue(value: Int): LogLevel? = when (value) {
            0 -> VERBOSE
            1 -> DEBUG
            2 -> INFO
            3 -> WARN
            4 -> ERROR
            else -> null
        }
    }
}
